requirements-full.txt

```diff
@@ -20,5 +21,5 @@ ndx-events==0.2.0
 parameterized==0.8.1
 pyintan==0.3.0
 pyopenephys==1.1.2
-sonpy
+sonpy>=1.7.1
```

Comment:
usually requirements-full specifies an exact version. Is that possible here?

Reply:
I was getting some dependency errors in my local installation when I tried pinning this to an exact version.

The problem is that is that the pip dependency resolver (or what it looks like it?) requires a different versions of `sonpy`  for python 3.7 and 3.9. 

I did not dig too deep into it but it might come from the flexibility allowed in `scipy` and `h5py`.

I could dig deeper into this if you think is worth it.

Reply:
oh yes I remember that being an issue. OK this is good

Reply:
I'm sure Ben remembers now but for for @h-mayorquin: `pip install sonpy` installs a different version depending on platform and Python version, and most of the cross-platform and cross-Python versions are not intercompatible. Attempts to pin to an exact version inevitably break the non-compatible regions of the test matrix, so we are forced to leave this as loose (at least on the upper bound). So this is fine.

Reply:
Yes. I realized when I was testing this. Thanks for the explanation!